pkg/cache/groupcache_test.go

```diff
@@ -52,30 +64,172 @@ func BenchmarkGroupcacheRetrieval(b *testing.B) {
 		log.NewJSONLogger(os.Stderr),
 		reg,
 		GroupcacheConfig{
-			Peers:           []string{selfURL},
-			SelfURL:         selfURL,
-			GroupcacheGroup: "groupName",
+			Peers:           []string{selfURLH1},
+			SelfURL:         selfURLH1,
+			GroupcacheGroup: groupName,
 			DNSSDResolver:   dns.MiekgdnsResolverType,
 			DNSInterval:     30 * time.Second,
 			MaxSize:         model.Bytes(16 * 1024),
 		},
-		"/_groupcache/",
+		basePath,
 		router,
 		bkt,
 		cachingBucketConfig,
 	)
-	testutil.Ok(b, err)
+	if err != nil {
+		fmt.Printf("failed creating group cache: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	go func() {
+		if err = httpServer.ListenAndServe(); err != nil {
+			fmt.Printf("failed to listen: %s\n", err.Error())
```

Comment:
We should do an `os.Exit(1)` here as well. If we fail to start the http server, we should fail the test here instead of continuing with it.